vault/resource_pki_secret_backend_key_test.go

```diff
@@ -0,0 +1,143 @@
+	resourceName := resourceType + ".test"
+
+	keyName := acctest.RandomWithPrefix("tf-pki-key")
+	updatedKeyName := acctest.RandomWithPrefix("tf-pki-key-updated")
+
+	store := &testPKIKeyStore{}
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		PreCheck: func() {
+			testutil.TestAccPreCheck(t)
+			SkipIfAPIVersionLT(t, testProvider.Meta(), provider.VaultVersion111)
+		},
+		CheckDestroy: testCheckMountDestroyed(resourceType, consts.MountTypePKI, consts.FieldMount),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPKISecretBackendKey_basic(mount, keyName, "rsa", "2048"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, consts.FieldMount, mount),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyName, keyName),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyType, "rsa"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyBits, "2048"),
+					testCapturePKIKeyID(resourceName, store),
+				),
+			},
+			// key name can be updated
+			// test that updating name ensures the key ID is unchanged
+			{
+				Config: testAccPKISecretBackendKey_basic(mount, updatedKeyName, "rsa", "2048"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, consts.FieldMount, mount),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyName, updatedKeyName),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyType, "rsa"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyBits, "2048"),
+					// ensure that the Key ID is unchanged
+					testPKIKeyUpdate(resourceName, store, true),
+					testCapturePKIKeyID(resourceName, store),
+				),
+			},
+			// any other updates trigger a force new
+			// test that key ID is different (new key created)
+			{
+				Config: testAccPKISecretBackendKey_basic(mount, updatedKeyName, "ec", "224"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, consts.FieldMount, mount),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyName, updatedKeyName),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyType, "ec"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyBits, "224"),
+					testPKIKeyUpdate(resourceName, store, false),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{consts.FieldType, consts.FieldKeyBits},
+			},
+		},
+	})
+}
+
+func testAccPKISecretBackendKey_basic(path, keyName, keyType, keyBits string) string {
+	return fmt.Sprintf(`
+resource "vault_mount" "pki" {
+	path        = "%s"
+	type        = "pki"
+    description = "PKI secret engine mount"
+}
+
+resource "vault_pki_secret_backend_key" "test" {
+  mount    = vault_mount.pki.path
+  type     = "exported"
```

Comment:
should we also add a test for type kms since we have some special handling for the kms fields?

Reply:
Good idea, I can add an additional test to confirm the extra fields work as expected

Reply:
Had some queries around how to configure a managed key in Vault. Will add this test in a follow-up PR after syncing w/ folks on the Crypto team 🙏🏼